app/models/host.rb

```
@@ -1413,6 +1351,69 @@ def scan_from_queue(taskid = nil)
     _log.info("Scanning #{log_target}...Complete - Timings: #{t.inspect}")
   end
 
+  def scan_via_ssh(task)
+    if hostname.blank?
+      _log.warn("No hostname defined for #{log_target}")
+      task.update_status("Finished", "Warn", "Scanning incomplete due to missing hostname")  if task
+      return
```

I think I need to raise an exception here so that caller can return, as before.